shared/src/main/scala/scala/util/parsing/input/OffsetPosition.scala

```diff
@@ -34,10 +34,14 @@ case class OffsetPosition(source: CharSequence, offset: Int) extends Position {
 
   private def genIndex: Array[Int] = {
     val lineStarts = new ArrayBuffer[Int]
-    lineStarts += 0
-    for (i <- 0 until source.length)
-      if (source.charAt(i) == '\n') lineStarts += (i + 1)
-    lineStarts += source.length
+    lineStarts += 0 // first line
+    for (i <- 0 until source.length) {
```

Comment:
may as well use `for i <- 1 until source.length` and remove the `if i>= 1` checks below